sql/core/src/test/scala/org/apache/spark/sql/execution/command/ShowFunctionsSuiteBase.scala

```diff
@@ -38,9 +38,11 @@ import org.apache.spark.util.Utils
 trait ShowFunctionsSuiteBase extends QueryTest with DDLCommandTestUtils {
   override val command = "SHOW FUNCTIONS"
 
+  protected def funCatalog: String = catalog
   protected def createFunction(name: String): Unit = {}
   protected def dropFunction(name: String): Unit = {}
   protected def showFun(ns: String, name: String): String = s"$ns.$name".toLowerCase(Locale.ROOT)
+  protected def isTempFunctions(): Boolean = false
```

Comment:
it's a bit weird to have this flag. Temp function does not belong to any catalog and should be tested in both v1 and v2 tests.

Reply:
I have added a test which checks temp functions in v2 impl. Would you mind if we improve test coverage in another PR since this PR aims to support v2 mostly.